### ISSUE TITLE
Two icons in the Dock (macOS)

### ISSUE THREAD

**Issue by lbariogl** [microsoft/vscode#138838]
Issue Type: <b>Bug</b>

After last update, whe I open VScode via command line two icons appear in the dock. Cheking in the activity monitor, the second (which keeps bouncing) should be related to the Extension Host: once I force-quitted the second instance I got the error "Extension host terminated unexpectedly".

VS Code version: Code 1.63.0 (Universal) (7db1a2b88f7557e0a43fec75b6ba7e50b3e9f77e, 2021-12-07T05:15:48.091Z)
OS version: Darwin x64 20.6.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-1068NG7 CPU @ 2.30GHz (8 x 2300)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 2, 2|
|Memory (System)|16.00GB (0.09GB free)|
|Process Argv|. --crash-reporter-id 9c5b4553-d61e-41f4-94d0-bc77eaf6ee21|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (40)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-languagetool|ada|3.8.0
root-file-viewer|alb|0.4.0
spellright|ban|3.0.60
bracket-pair-colorizer|Coe|1.0.61
doxdocgen|csc|1.3.2
dart-code|Dar|3.29.0
flutter|Dar|3.29.0
gitlens|eam|11.7.0
vscode-pull-request-github|Git|0.34.0
latex-workshop|Jam|8.22.0
better-cpp-syntax|jef|1.15.10
vscode-clang|mit|0.2.4
vscode-docker|ms-|1.18.0
python|ms-|2021.12.1559732655
vscode-pylance|ms-|2021.12.1
jupyter|ms-|2021.11.1001550889
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.4
remote-containers|ms-|0.209.3
remote-ssh|ms-|0.68.0
remote-ssh-edit|ms-|0.68.0
remote-wsl|ms-|0.63.4
vscode-remote-extensionpack|ms-|0.21.0
atom-keybindings|ms-|3.0.9
cmake-tools|ms-|1.9.2
cpptools|ms-|1.8.0-insiders
cpptools-extension-pack|ms-|1.1.0
vsliveshare|ms-|1.0.5196
vsliveshare-audio|ms-|0.1.91
debugger-for-chrome|msj|4.13.0
postscript|mxs|1.0.1
dart-syntax-highlighting-only|osc|1.0.1
material-icon-theme|PKi|4.11.0
quicktype|qui|12.0.46
vscode-commons|red|0.0.6
vscode-yaml|red|1.2.2
bootstrap4-vscode|the|6.1.0
latex-support|tor|3.10.0
cmake|twx|0.0.17
vscode-todo-highlight|way|1.0.5

(3 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805:30301674
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
pythonrunftest32:30373476
pythonf5test824:30373475
javagetstartedt:30391933
pythonvspyt187:30373474
vsaa593:30376534
vsena315cf:30406436
pythonvs932:30410667
vscop804:30404766
vscop453:30404998
vsrem710:30405998
vscexrecpromp3t3:30407764

```

</details>

**Comment by adrum** [comment]
I'm also seeing multiple icons. Strangely, I had to quit them independently and my main VSCode window reported "Extension host terminated unexpectedly."

**Comment by tylersticka** [comment]
After disabling all extensions, this problem went away for me. By manually re-enabling them one by one, I narrowed the problem down to the `spellright` extension, which seems to have an open issue: https://github.com/bartosz-antosik/vscode-spellright/issues/465

**Comment by hesselmonk** [comment]
Why does VSCode even allow extensions to add another icon to the dock? To me this sounds like a VSCode issue exposed by extensions, not the other way around.

**Comment by bhagatparwinder** [comment]
> After disabling all extensions, this problem went away for me. By manually re-enabling them one by one, I narrowed the problem down to the `spellright` extension, which seems to have an open issue: [bartosz-antosik/vscode-spellright#465](https://github.com/bartosz-antosik/vscode-spellright/issues/465)

The whole situation was frustrating to resolve! You did the leg work. Kudos!

**Comment by 1deadpixl** [comment]
> After disabling all extensions, this problem went away for me. By manually re-enabling them one by one, I narrowed the problem down to the spellright extension, which seems to have an open issue: bartosz-antosik/vscode-spellright#465

@tylersticka Thank you! This has been bugging me for a while. I had like 4 or 5 VSC icons in my dock all the time. Was about to file a bug report today after seeing all the "duplicates" about launching `code` from terminal which wasn't my problem. In the process of filing the report, I diagnosed it to coming from an extension and so glad I found this issue before having to go through my 60-odd extensions!

> Why does VSCode even allow extensions to add another icon to the dock? To me this sounds like a VSCode issue exposed by extensions, not the other way around.

@hesselmonk Also this! I imagine there are very few use cases for extensions adding a dock icon. Seems like it's only an issue on MacOS right now (from https://github.com/bartosz-antosik/vscode-spellright/issues/465#issuecomment-995771625) so possibly something to be resolved in the interaction between VSC and MacOS.

I appreciate the community coming together to resolve this. Thank you!
